src/template-tags/tickets.php

```diff
@@ -472,193 +483,202 @@ function tribe_tickets_get_ticket_stock_message( Tribe__Tickets__Ticket_Object $
+		foreach ( $templates as $template ) {
+			$file = Tribe__Tickets__Templates::get_template_hierarchy( $template, array( 'disable_view_check' => true ) );
+
+			/**
+			 * Allow users to filter which template will be included
+			 *
+			 * @param string $file     Complete path to include the PHP File
+			 * @param string $template The Template file, which is a relative path from the Folder we are dealing with
+			 * @param string $slug     Slug for this template
+			 * @param string $name     Template name
+			 * @param array  $data     The Data that will be used on this template
+			 */
+			$file = apply_filters( 'tribe_tickets_get_template_part_path', $file, $template, $slug, $name, $data );
+
+			/**
+			 * A more Specific Filter that will include the template name
+			 *
+			 * @param string $file Complete path to include the PHP File
+			 * @param string $slug Slug for this template
+			 * @param string $name Template name
+			 * @param array  $data The Data that will be used on this template
+			 */
+			$file = apply_filters( "tribe_tickets_get_template_part_path_{$template}", $file, $slug, $name, $data );
+
+			if ( ! file_exists( $file ) ) {
+				continue;
+			}
 
-		if ( $echo ) {
-			echo $html;
-		}
+			ob_start();
+			/**
+			 * Fires an Action before including the template file
+			 *
+			 * @param string $template The Template file, which is a relative path from the Folder we are dealing with
+			 * @param string $file     Complete path to include the PHP File
+			 * @param string $slug     Slug for this template
+			 * @param string $name     Template name
+			 * @param array  $data     The Data that will be used on this template
+			 */
+			do_action( 'tribe_tickets_before_get_template_part', $template, $file, $slug, $name, $data );
+			include( $file );
+
+			/**
+			 * Fires an Action After including the template file
+			 *
+			 * @param string $template The Template file, which is a relative path from the Folder we are dealing with
+			 * @param string $file     Complete path to include the PHP File
+			 * @param string $slug     Slug for this template
+			 * @param string $name     Template name
+			 * @param array  $data     The Data that will be used on this template
+			 */
+			do_action( 'tribe_tickets_after_get_template_part', $template, $file, $slug, $name, $data );
+			$html = ob_get_clean();
+
+			/**
+			 * Allow users to filter the final HTML
+			 *
+			 * @param string $html     The final HTML
+			 * @param string $template The Template file, which is a relative path from the Folder we are dealing with
+			 * @param string $file     Complete path to include the PHP File
+			 * @param string $slug     Slug for this template
+			 * @param string $name     Template name
+			 * @param array  $data     The Data that will be used on this template
+			 */
+			$html = apply_filters( 'tribe_tickets_get_template_part_content', $html, $template, $file, $slug, $name, $data );
+
+			if ( $echo ) {
+				echo $html;
```

Comment:
:warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$html'

<sub>hash: 2cace009d64e98c07c52e7569acfcb0c</sub>